src/services/services.ts

```diff
@@ -2788,18 +2788,37 @@ namespace ts {
         return node && node.parent && node.parent.kind === SyntaxKind.PropertyAccessExpression && (<PropertyAccessExpression>node.parent).name === node;
     }
 
+    function climbPastPropertyAccess(node: Node) {
+        return isRightSideOfPropertyAccess(node) ? node.parent : node;
+    }
+
     function isCallExpressionTarget(node: Node): boolean {
-        if (isRightSideOfPropertyAccess(node)) {
-            node = node.parent;
-        }
-        return node && node.parent && node.parent.kind === SyntaxKind.CallExpression && (<CallExpression>node.parent).expression === node;
+        return !!getCallOrNewExpressionWorker(node, SyntaxKind.CallExpression);
     }
 
     function isNewExpressionTarget(node: Node): boolean {

```

Comment:
The semantics of these functions are different now. They no longer only check if their operand is the target of the call. Instead, they climb up from any given member. Is this behavior intended?

If you don't need to change their behavior, I'd rather you just made `getCallOrNewExpressionWorker` take a single node. Then make `getCallOrNewTargetAncestor` pass it the result of calling `climbPastPropertyAccess`.

That way, you also save two climbs (one for `CallExpression` and one for `NewExpression`)